### DISCUSSION TITLE
issue: Validate is not firing, trying to confirm password

### DISCUSSION THREAD

**Discussion by lewisd1996** [react-hook-form/react-hook-form#7324] [Q&A]
### Version Number

7.22.1

### Codesandbox/Expo snack

https://codesandbox.io/s/inspiring-haslett-1c8sw?file=/pages/index.js

### Steps to reproduce

1. Enter a password into the first field
2. Try match it in the second field
3. The error message will never show, it will just say "Passwords match"

### Expected behaviour

The "Passwords do not match" message should appear instead of "Passwords match" message, until the passwords match of course

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
resources: 

- https://jasonwatmore.com/post/2021/09/15/react-hook-form-password-and-confirm-password-match-validation-example
- https://codesandbox.io/s/react-hook-form-password-match-check-standard-validation-eo6en?file=/src/index.js:0-39 (v6 but can easily convert to v7)

**Reply by lewisd1996** [reply]
Thanks for reply bill, so regarding the first link, I need to convert my hook form to this other version that uses `yup` explicitly? 

I was under the impression that in v7, this was the correct way to add validation: (I am using `watch` to watch the value of `password`)

`{...register("confirmPassword", {
      validate: (value) => value === password
 })}`
